server/routes/api/documents/documents.ts

```diff
@@ -1975,9 +1976,8 @@ router.post(
       paranoid: false,
     });
 
-    await documentPermanentDeleter(documents);
-    await Event.createFromContext(ctx, {
-      name: "documents.empty_trash",
+    await EmptyTrashTask.schedule({
+      documentIds: documents.map((doc) => doc.id),
```

Comment:
API determines the documents to delete at the time it was invoked by the user.

Reply:
Good call

Reply:
We could use `attributes` on the initial query to only retrieve the `id` and speed things up further